stan/math/opencl/prim/normal_id_glm_lpdf.hpp

```diff
@@ -152,41 +143,63 @@ return_type_t<T_alpha, T_beta, T_scale> normal_id_glm_lpdf(
                     calc_if<need_log_sigma_sum>(log_sigma_sum_expr));
 
   double y_scaled_sq_sum = sum(from_matrix_cl(y_scaled_sq_sum_cl));
-  operands_and_partials<T_alpha_ref, T_beta_ref, T_sigma_ref> ops_partials(
-      alpha_ref, beta_ref, sigma_ref);
-  if (!is_constant_all<T_alpha>::value && is_alpha_vector) {
-    ops_partials.edge1_.partials_
-        = from_matrix_cl<Dynamic, 1>(mu_derivative_cl);
-  }
+  operands_and_partials<T_y, T_x, T_alpha, T_beta, T_scale> ops_partials(
+      y, x, alpha, beta, sigma);
+  double mu_derivative_sum;
   if (need_mu_derivative_sum) {
-    ops_partials.edge1_.partials_[0]
-        = sum(from_matrix_cl(mu_derivative_sum_cl));
+    mu_derivative_sum = sum(from_matrix_cl(mu_derivative_sum_cl));
   }
-  if (!is_constant_all<T_beta>::value) {
-    const matrix_cl<double> mu_derivative_transpose_cl(
-        mu_derivative_cl.buffer(), 1,
-        mu_derivative_cl.rows());  // transposition of a vector can be done
-    // without copying
+  if (!is_constant_all<T_y>::value) {
```

Comment:
```suggestion
  if (!is_constant<T_y>::value) {
```

Reply:
Done. I don't know why all the distributions are using the `_all` variant for single argument ...